cnd/src/main.rs

```diff
@@ -52,6 +53,8 @@ fn main() -> anyhow::Result<()> {
 
     crate::trace::init_tracing(settings.logging.level)?;
 
+    let _file = lock_datadir(&settings.data.dir)?;
```

Comment:
Does it print an error on stderr on failure?

Reply:
No idea tbh. @thomaseizinger  do you know where this error will be printed?

Reply:
Should I put an exclusive lock on the config file as well or not? You said it will crash anyway due to a port binding failure?

Reply:
Lock on data dir is enough.
Just try the behaviour locally ;)